instrumentation/openai-client-instrumentation/common/src/main/java/co/elastic/otel/openai/wrappers/InstrumentedChatCompletionService.java

```diff
@@ -137,11 +137,9 @@ public void onStart(
                       .responseFormat()
                       .ifPresent(
                           val -> {
-                            if (val.isResponseFormatText()) {
-                              attributes.put(
-                                  GEN_AI_OPENAI_REQUEST_RESPONSE_FORMAT,
-                                  val.asResponseFormatText()._type().toString());
-                            }
+                            attributes.put(
+                                GEN_AI_OPENAI_REQUEST_RESPONSE_FORMAT,
+                                ApiAdapter.get().extractType(val));
```

Comment:
If I remember correctly adding an attribute with a `null` value is ignored, so if the response is not text then we should not add anything, is that correct ?

Reply:
I checked the implementation, you are correct. Nonetheless we should introduce a null check I think for clarity and make `extractType` return `null` if it can't extract a type.

> so if the response is not text then we should not add anything, is that correct
It should return a string identifying the type whenever it is available. The previous implementation did this only for text, so this is a slight enhancement I smuggled into this PR.

Reply:
Done in [106a90c](https://github.com/elastic/elastic-otel-java/pull/531/commits/106a90c8e32b287f761aa168da49ba6077a73823).